File: src/main/java/com/wgzhao/addax/admin/repository/UserRepo.java
package com.wgzhao.addax.admin.repository;

import com.wgzhao.addax.admin.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepo extends JpaRepository<User, Long> {

    User findByEmail(String email);

    User findByUsername(String username);

    boolean existsByUsername(String username);
}

File: src/main/kotlin/com/wgzhao/addax/admin/repository/UserRepo.kt
package com.wgzhao.addax.admin.repository

import com.wgzhao.addax.admin.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepo : JpaRepository<User?, Long?> {
    fun findByEmail(email: String?): User?

    fun findByUsername(username: String?): User?

    fun existsByUsername(username: String?): Boolean
}
